contracts/contracts/multisig/Multisig.cairo

```diff
@@ -474,24 +436,113 @@ func execute_transaction{
     return (response_len=response.retdata_size, response=response.retdata)
 end
 
+# Sets number of required confirmations. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_confirmations_required.
+@external
+func set_confirmations_required{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    confirmations_required : felt
+):
+    const lower_bound = 1
+
+    let (caller) = get_caller_address()
+    let (contract_address) = get_contract_address()
+
+    with_attr error_message("caller shall be multisig"):
+        assert caller = contract_address
+    end
+
+    let (owners_len) = _owners_len.read()
+
+    with_attr error_message("invalid number of required confirmations"):
+        assert_in_range(confirmations_required, lower_bound, owners_len + 1)
+    end
+
+    _confirmations_required.write(confirmations_required)
+    ConfirmationsSet.emit(confirmations_required)
```

Comment:
Also used in the constructor, where the event is not emitted.
Consider extracting as an internal fn and reuse.

Reply:
Done.